apps/src/aichat/redux/utils.ts

```diff
@@ -29,6 +29,10 @@ const haveDifferentValues = (
   if (typeof value1 === 'object' && typeof value2 === 'object') {
     return JSON.stringify(value1) !== JSON.stringify(value2);
   }
+  // In the case that field values are saved as different types, compare as strings.
```

Comment:
Since we currently have mixed types for `temperature` in .level files and sources, we include this failsafe just in case.

Reply:
Nice!